src/zenml/models/v2/core/flavor.py

```diff
@@ -0,0 +1,317 @@
+        title="The JSON schema of this flavor's corresponding configuration.",
+    )
+    connector_type: Optional[str] = Field(
+        default=None,
+        title="The type of the connector that this flavor uses.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    connector_resource_type: Optional[str] = Field(
+        default=None,
+        title="The resource type of the connector that this flavor uses.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    connector_resource_id_attr: Optional[str] = Field(
+        default=None,
+        title="The name of an attribute in the stack component configuration "
+        "that plays the role of resource ID when linked to a service "
+        "connector.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    source: str = Field(
+        title="The path to the module which contains this Flavor.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    integration: Optional[str] = Field(
+        title="The name of the integration that the Flavor belongs to.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    logo_url: Optional[str] = Field(
+        default=None,
+        title="Optionally, a url pointing to a png,"
+        "svg or jpg can be attached.",
+    )
+    docs_url: Optional[str] = Field(
+        default=None,
+        title="Optionally, a url pointing to docs, within docs.zenml.io.",
+    )
+    sdk_docs_url: Optional[str] = Field(
+        default=None,
+        title="Optionally, a url pointing to SDK docs,"
+        "within sdkdocs.zenml.io.",
+    )
+    is_custom: bool = Field(
+        title="Whether or not this flavor is a custom, user created flavor.",
+        default=True,
+    )
+    user: Optional[UUID] = Field(
+        default=None, title="The id of the user that created this resource."
+    )
+    workspace: Optional[UUID] = Field(
+        default=None, title="The workspace to which this resource belongs."
+    )
+
+
+# ------------------ Update Model ------------------
+
+
+@update_model
+class FlavorUpdate(FlavorRequest):
+    """Update model for flavors."""
+
+
+# ------------------ Response Model ------------------
+
+
+class FlavorResponseBody(BaseResponseBody):
+    """Response body for flavor."""
+
+    user: Union["UserResponse", None] = Field(
+        title="The user that created this resource.", nullable=True
+    )
+    type: StackComponentType = Field(title="The type of the Flavor.")
+    integration: Optional[str] = Field(
+        title="The name of the integration that the Flavor belongs to.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    logo_url: Optional[str] = Field(
+        default=None,
+        title="Optionally, a url pointing to a png,"
+        "svg or jpg can be attached.",
+    )
+
+
+class FlavorResponseMetadata(BaseResponseMetadata):
+    """Response metadata for flavors."""
+
+    workspace: Optional["WorkspaceResponse"] = Field(
+        title="The project of this resource."
+    )
```

Comment:
For some reason unknown to me, the flavors were originally designed by inheriting from the base response models instead of workspace-scoped models. If you look closely, you will also see while the workspace-scoped response models had a mandatory `workspace_id` and an optional `user_id`, the flavors had both fields optional. This is why I stuck true to the original design and kept the same fields.

Reply:
It's gonna be removed soonish anyways so whatever I guess :D